TLM/TLM/Manager/Impl/JunctionRestrictionsManager.cs

```diff
@@ -13,492 +13,481 @@
-        }
+			return ret;
+		}
 
-        public bool GetDefaultUturnAllowed(ushort segmentId, bool startNode, ref NetNode node) {
+		public bool GetDefaultUturnAllowed(ushort segmentId, bool startNode, ref NetNode node) {
 #if DEBUG
-            bool debug = GlobalConfig.Instance.Debug.Switches[11];
+			bool debug = GlobalConfig.Instance.Debug.Switches[11];
 #endif
 
-            if (!Constants.ManagerFactory.JunctionRestrictionsManager.IsUturnAllowedConfigurable(segmentId, startNode, ref node)) {
-                bool res = (node.m_flags & (NetNode.Flags.End | NetNode.Flags.OneWayOut)) != NetNode.Flags.None;
+			if (!Constants.ManagerFactory.JunctionRestrictionsManager.IsUturnAllowedConfigurable(segmentId, startNode, ref node)) {
+				bool res = (node.m_flags & (NetNode.Flags.End | NetNode.Flags.OneWayOut)) != NetNode.Flags.None;
 #if DEBUG
-                if (debug)
-                    Log._Debug($"JunctionRestrictionsManager.GetDefaultUturnAllowed({segmentId}, {startNode}): Setting is not configurable. res={res}, flags={node.m_flags}");
+				if (debug)
+					Log._Debug($"JunctionRestrictionsManager.GetDefaultUturnAllowed({segmentId}, {startNode}): Setting is not configurable. res={res}, flags={node.m_flags}");
 #endif
-                return res;
-            }
+				return res;
+			}
 
-            bool ret = (node.m_flags & (NetNode.Flags.End | NetNode.Flags.OneWayOut)) != NetNode.Flags.None;
+			bool ret = (node.m_flags & (NetNode.Flags.End | NetNode.Flags.OneWayOut)) != NetNode.Flags.None;
 
-            if (!ret && Options.allowUTurns) {
-                ret = (node.m_flags & (NetNode.Flags.Junction | NetNode.Flags.Transition)) != NetNode.Flags.None;
-            }
+			if (!ret && Options.allowUTurns) {
+				ret = (node.m_flags & (NetNode.Flags.Junction | NetNode.Flags.Transition)) != NetNode.Flags.None;
+			}
 
 #if DEBUG
-            if (debug)
-                Log._Debug($"JunctionRestrictionsManager.GetDefaultUturnAllowed({segmentId}, {startNode}): Setting is configurable. ret={ret}, flags={node.m_flags}");
+			if (debug)
+				Log._Debug($"JunctionRestrictionsManager.GetDefaultUturnAllowed({segmentId}, {startNode}): Setting is configurable. ret={ret}, flags={node.m_flags}");
 #endif
 
-            return ret;
-        }
+			return ret;
+		}
 
-        public bool IsUturnAllowed(ushort segmentId, bool startNode) {
-            return SegmentFlags[segmentId].IsUturnAllowed(startNode);
-        }
+		public bool IsUturnAllowed(ushort segmentId, bool startNode) {
+			return SegmentFlags[segmentId].IsUturnAllowed(startNode);
+		}
 
-        public bool IsTurnOnRedAllowedConfigurable(ushort segmentId, bool startNode, ref NetNode node) {
+		public bool IsTurnOnRedAllowedConfigurable(ushort segmentId, bool startNode, ref NetNode node) {
 #if DEBUG
-            bool debug = GlobalConfig.Instance.Debug.Switches[11];
+			bool debug = GlobalConfig.Instance.Debug.Switches[11];
 #endif
-	        if (!Options.turnOnRed)
-		        return false;
```

Comment:
I don't think a feature toggle is required. The feature does not have an impact on performance. You have another flag that can be used to toggle the feature.

Reply:
Feature toggle restored